userspace/libsinsp/filter.cpp

```diff
@@ -2418,5 +2418,48 @@ gen_event_filter_check *sinsp_filter_factory::new_filtercheck(const char *fldnam
 							  true);
 }
 
+std::list<gen_event_filter_factory::filter_fieldclass_info> sinsp_filter_factory::get_fields()
+{
+	std::list<gen_event_filter_factory::filter_fieldclass_info> ret;
+
+	vector<const filter_check_info*> fc_plugins;
+	sinsp::get_filtercheck_fields_info(&fc_plugins);
+
+	for(uint32_t j = 0; j < fc_plugins.size(); j++)
+	{
+		const filter_check_info* fci = fc_plugins[j];
+
+		if(fci->m_flags & filter_check_info::FL_HIDDEN)
+		{
+			continue;
+		}
+
+		gen_event_filter_factory::filter_fieldclass_info cinfo;
+		cinfo.name = fci->m_name;
+		cinfo.desc = "";
+		cinfo.class_info = "";
+
+		for(int32_t k = 0; k < fci->m_nfields; k++)
+		{
+			const filtercheck_field_info* fld = &fci->m_fields[k];
+
+			if(fld->m_flags & EPF_TABLE_ONLY ||
+			   fld->m_flags & EPF_PRINT_ONLY)
+			{
+				continue;
+			}
```

Comment:
What do these flags mean and why do we want this check here?

Reply:
They are defined here: https://github.com/falcosecurity/libs/blob/factory-add-get-fields/userspace/libsinsp/event.h#L44. Generally if a field can't be used to filter events, we don't want to print it.

(And btw, this is just moving code from falco that did this in a specific way for syscall filterchecks. This PR makes it generic for a "factory" e.g. source of filters/filterchecks/fields).

Reply:
Thanks for the explanation.

> if a field can't be used to filter events, we don't want to print it

this should live as a comment above this `if`

Reply:
Okay, added.